sdk/dapp-kit/src/components/SuiClientProvider.tsx

```diff
@@ -45,13 +46,20 @@ export function SuiClientProvider<T extends NetworkConfigs>(props: SuiClientProv
 		(props.createClient as typeof DEFAULT_CREATE_CLIENT) ?? DEFAULT_CREATE_CLIENT;
 
 	const [selectedNetwork, setSelectedNetwork] = useState<keyof T & string>(
-		props.defaultNetwork ?? (Object.keys(networks)[0] as keyof T & string),
+		props.network ?? props.defaultNetwork ?? (Object.keys(networks)[0] as keyof T & string),
 	);
 
 	const [client, setClient] = useState<SuiClient>(() => {
 		return createClient(selectedNetwork, networks[selectedNetwork]);
 	});
 
+	useEffect(() => {
+		if (props.network && props.network !== selectedNetwork) {
+			setSelectedNetwork(props.network);
+			setClient(createClient(props.network, networks[props.network]));
+		}
+	}, [createClient, networks, props.network, selectedNetwork, setClient, setSelectedNetwork]);
```

Comment:
this is hacking around this currently supporting both passing the selected network via props and managing the current network via the context.  I don't think we should support both in the same provider.